src/hover_provider/sourceHoverProvider.ts

```diff
@@ -123,6 +123,31 @@ export class SourceHoverProvider implements HoverProvider, Disposable {
       content.appendMarkdown(
         `<span style="color:#347890;">(source)&nbsp;</span><span><strong>${node.name}</strong></span>`,
       );
+      if (node.description !== "") {
+        content.appendMarkdown(`</br><span>${node.description}</span>`);
+      }
+      content.appendText("\n");
```

Comment:
We are adding a lot of new lines, however if either description or columns are not present, this may not be presented very beautifully. 

Could you ensre that the lines are only added if the relevant section is present?

Reply:
As this issue is also present in the model code. Maybe you can refactor this out to a separate function and use this function in both places?

Reply:
The newlines actually don't render as newlines the way it looks. They are just there so that the `---` is shown as a line. Like this:

---

When the descriptions or columns aren't present the hover box looks like:
<img width="355" alt="Screenshot 2023-09-26 at 8 36 13 AM" src="https://github.com/AltimateAI/vscode-dbt-power-user/assets/79863718/e9d5e7cf-f10a-4644-b0e3-1cd43d7059c4">
I can still remove it if columns aren't present. It is straight forward enough to change.

In any case I can definitely consolidate the code into a single function before getting this merged. They are nearly identical.

Reply:
Great! Thanks for clarifying.